File: fluorender/Linux/launcher.py
#!/usr/bin/python3
import sys, os
if getattr(sys, 'frozen', False):
    bundle_dir = sys._MEIPASS
else:
    bundle_dir = os.path.dirname(os.path.abspath(__file__))

print( 'bundle dir is', bundle_dir )
print( 'sys.executable is', os.path.dirname(sys.executable) )

print(f"Arguments count: {len(sys.argv)}")
for i, arg in enumerate(sys.argv):
    print(f"Argument {i:>6}: {arg}")

dir = os.path.dirname(sys.executable)

command = "export GDK_PIXBUF_MODULE_FILE=" + dir + "/data/loaders.cache" + " && "
command += "export XDG_DATA_DIRS=\"" + dir + "/data:" + dir + "/data/schemas:" + "/usr/shre/gnome:/usr/local/share:/usr/share\"" + " && "
command += dir + "/data/VVDViewer "
for i, arg in enumerate(sys.argv):
    if i > 0:
        command += arg + " "

os.system(command)

File: fluorender/Linux/launcher.cpp
#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>
#include <libgen.h>
#include <linux/limits.h>
#include <filesystem>
#include <sstream>
namespace fs = std::filesystem;

int main()
{
    char exepath[PATH_MAX];
	ssize_t count = readlink("/proc/self/exe", exepath, PATH_MAX);
	const char *exedir;
	if (count != -1) {
    	exedir = dirname(exepath);
	}

    fs::path dir (exedir);
    fs::path schema_dir ("data/schemas");
    fs::path cache_file ("data/loaders.cache");
    fs::path exe_file ("data/VVDViewer");
    fs::path ld_file ("data/lib/ld-linux-x86-64.so.2");
    fs::path cache_full_path = dir / cache_file;
    fs::path schemas_full_path = dir / schema_dir;
    fs::path exe_full_path = dir / exe_file;
    fs::path ld_full_path = dir / ld_file;

    std::stringstream ss;

    ss << "export GDK_PIXBUF_MODULE_FILE=" << cache_full_path << " && " << "export XDG_DATA_DIRS=\"" << schemas_full_path << "/usr/shre/gnome:/usr/local/share:/usr/share\" && " << ld_full_path << " " << exe_full_path;

    system(ss.str().c_str());
}